File: LeetCode/0346.Moving-Average-From-Data-Stream/Moving-Average-From-Data-Stream.java
public class MovingAverage {
    /*
    * @param size: An integer
    
    */
    private Queue<Integer> queue ;
    private int size;
    private double sum = 0;
    public MovingAverage(int size) {
        // do intialization if necessary
        this.size = size;
        queue = new LinkedList<>();
    }

    /*
     * @param val: An integer
     * @return:  
     */
    public double next(int val) {
        // write your code here
        sum = sum + val;
        if(queue.size() == size){
            sum = sum - queue.poll();
        }
        queue.offer(val);
        return (double) sum / queue.size();
    }
}

File: LeetCode/0346.Moving-Average-From-Data-Stream/Moving-Average-From-Data-Stream.py
from collections import deque
class MovingAverage(object):

    def __init__(self, size):
        """
        Initialize your data structure here.
        :type size: int
        """
        self.size = size
        self.queue = deque()
        self.sum = 0

    def next(self, val):
        """
        :type val: int
        :rtype: float
        """
        if len(self.queue) == self.size:
            self.sum -= self.queue.popleft()
        self.sum += val
        self.queue.append(val)
        return self.sum / float(len(self.queue))
        


# Your MovingAverage object will be instantiated and called as such:
# obj = MovingAverage(size)
# param_1 = obj.next(val)
